Library/SubscriptionChange.cs

```diff
@@ -0,0 +1,175 @@
+    /// </summary>
+    public class SubscriptionChange
+    {
+        public enum ChangeTimeframe : short
+        {
+            Now,
+            Renewal
+        }
+
+        public ChangeTimeframe TimeFrame { get; set; }
+
+        public string PlanCode { get; set; }
+
+        /// <summary>
+        /// List of custom fields
+        /// </summary>
+        public List<CustomField> CustomFields
+        {
+            get { return _customFields ?? (_customFields = new List<CustomField>()); }
+            set { _customFields = value; }
+        }
+        private List<CustomField> _customFields;
+
+        public string Uuid { get; set; }
+
+        /// <summary>
+        /// Unit amount per quantity.  Leave null to keep as is. Set to override plan's default amount.
+        /// </summary>
+        public int? UnitAmountInCents { get; set; }
+
+        public int? Quantity { get; set; }
+
+        /// <summary>
+        /// List of add ons for this subscription
+        /// </summary>
+        public SubscriptionAddOnList AddOns
+        {
+            get { return _addOns ?? (_addOns = new SubscriptionAddOnList(new Subscription())); }
+            set { _addOns = value; }
+        }
+
+        private SubscriptionAddOnList _addOns;
+        /// <summary>
+        /// The coupon code you want to redeem in the update.
+        /// Only allowed if timeframe is now and you change something about the subscription that creates an invoice.
+        /// </summary>
+        public string CouponCode { get; set; }
+
+        public int? RemainingBillingCycles { get; set; }
+
+        public string CollectionMethod { get; set; }
+
+        public int? NetTerms { get; set; }
+
+        public string PoNumber { get; set; }
+
+        /// <summary>
+        /// Determines whether subscriptions to this plan should auto-renew term at the end of the current term or expire.
+        /// Defaults to true.
+        /// </summary>
+        public bool? AutoRenew { get; set; }
+
+        /// <summary>
+        /// Determines the renewal subscription term.
+        /// Defaults to plans total billing cycles value unless
+        /// overwritten when creating the subscription or editing subscription.
+        /// </summary>
+        public int? RenewalBillingCycles { get; set; }
+
+        public Adjustment.RevenueSchedule? RevenueScheduleType { get; set; }
+
+        /// <summary>
+        /// Optionally set true to indicate if the subscription is to be treated as an import
+        /// for data analysis or a real trial. This value can only be changed on a trial subscription
+        /// and is persisted for its lifetime.
+        /// </summary>
+        public bool? ImportedTrial { get; set; }
+
+        /// <summary>
+        /// Creates a new subscription change object
+        /// </summary>
+        /// <param name="uuid">The uuid of the subscription to be changed</param>
+        public SubscriptionChange(String uuid)
```

Comment:
I like the idea of making it required, but I think it might make more sense to make it a parameter on the method.

```csharp
var change = new SubscriptionChange() {
    PlanCode = "gold"
};
var uuid = "48e99881b3726425bf49e64b45af7fe4";
var newSub = Subscription.PreviewChange(uuid, change);
```

I feel like I could go either way on it. What do you think?

Reply:
I think the reason my gut is telling me to go this way is that it's a URL parameter.

Reply:
I think that you're right, but perhaps for a different reason. The idea of the `SubscriptionChange` class is that each instance of the class would represent the set of changes that need to be sent to the server.

The UUID isn't something that is going to change, it's just something necessary for the endpoint. For that reason, I feel like `Uuid` shouldn't even be a property of the SubscriptionChange class. I think passing it into the method would make a lot of sense

Reply:
Yeah, let's do that.